packages/viz/src/lib/Layer.ts

```diff
@@ -1,164 +1,133 @@
-
-    this._layerSource = new Source(source);
-    this._layerStyles = new Style(styles);
-
-    this._layerOptions = { ...defaultMapOptions, ...mapOptions };
-    this._layerInstantiation = this._mapsClientInstance.instantiateMapFrom(
-      buildInstantiationOptions({
-        mapOptions: this._layerOptions,
-        mapSource: this._layerSource
-      })
-    );
+  constructor(source: string | Source, styles = {}) {
+    this._source = buildSource(source);
+    this._styles = new Style(styles);
   }
 
-  public async setSource(source: string) {
-    const previousSource = this._layerSource;
+  /**
+   * Change a source to the current layer.
+   * A new map instantion and a replace of the layer will be fired
+   * @param source source to be set
+   */
+  public async setSource(source: string | Source) {
+    const previousSource = this._source;
 
-    this._layerSource = new Source(source);
-    this._layerInstantiation = this._mapsClientInstance.instantiateMapFrom(
-      buildInstantiationOptions({
-        mapOptions: this._layerOptions,
-        mapSource: this._layerSource
-      })
-    );
+    this._source = buildSource(source);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckLayer) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Change the styles of the current layer.
+   * A new map instantion and a replace of the layer will be fired
+   * @param style style to be set
+   */
   public async setStyle(style: {}) {
-    const previousSource = this._layerSource;
+    const previousSource = this._source;
 
-    this._layerStyles = new Style(style);
+    this._styles = new Style(style);
 
-    if (this._mvtLayerInstance) {
+    if (this._deckLayer) {
       await this._replaceLayer(previousSource);
     }
   }
 
+  /**
+   * Add the current layer to a Deck instance
+   * @param deckInstance instance to add the layer to
+   */
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   public async addTo(deckInstance: any) {
     const currentDeckLayers = deckInstance.props.layers;
-    const createdDeckLayer = await this.getDeckGLLayer();
+    const layer = await this._createDeckGLLayer();
 
     deckInstance.setProps({
-      layers: [...currentDeckLayers, createdDeckLayer]
+      layers: [...currentDeckLayers, layer]
     });
 
     this._deckInstance = deckInstance;
   }
 
-  public async getDeckGLLayer() {
-    const layerProperties = await this._getDeckGLLayerProperties();
-
-    this._mvtLayerInstance = new MVTLayer(layerProperties);
-    return this._mvtLayerInstance;
-  }
-
-  private async _getDeckGLLayerProperties() {
-    const {
-      urlTemplates: data,
-      geometryType
-    } = await this._layerInstantiation.then(parseInstantiationResult);
+  /**
+   * Method to create the Deck.gl layer
+   */
+  public async _createDeckGLLayer() {
+    // Get properties of the layer
+    const props = await this._source.getLayerProps();
```

Comment:
I would think about a `getMetadata`, instead of `getLayerProps`. That data could be useful for other things, beyond layer